src/main/java/org/mule/extension/runtime/OperationContext.java

```diff
@@ -44,4 +48,22 @@
      * @return a {@code C} consistent with a corresponding {@link Configuration} model
      */
     <C> C getConfigurationInstance();
+
+    /**
+     * Registers a {@link Consumer} which takes a {@link OperationSuccessfulEvent}
+     * and performs an action if the operation associated with {@code this} instance
+     * is executed successfully
+     *
+     * @param handler a {@link Consumer} which handles a {@link OperationSuccessfulEvent}
+     */
+    void onOperationSuccessful(Consumer<OperationSuccessfulEvent> handler);
```

Comment:
This method is just to add the "event listener"... The listener itself is in the implementation of the api, not the api itself